docs/bigquery/snippets.py

```diff
@@ -2073,5 +2077,44 @@ def test_list_rows_as_dataframe(client):
     assert len(df) == table.num_rows           # verify the number of rows
 
 
+@pytest.mark.skipIf(pandas is None, reason='Requires `pandas`')
+@pytest.mark.skipIf(pyarrow is None, reason='Requires `pyarrow`')
+def test_load_table_from_dataframe(client, to_delete):
+    dataset_id = 'load_table_dataframe_dataset_{}'.format(_millis())
+    dataset = bigquery.Dataset(client.dataset(dataset_id))
+    client.create_dataset(dataset)
+    to_delete.append(dataset)
+
+    # [START bigquery_load_table_dataframe]
+    # from google.cloud import bigquery
+    # client = bigquery.Client()
+    # dataset_id = 'my_dataset'
+
+    dataset_ref = client.dataset(dataset_id)
+    table_ref = dataset_ref.table('monty_python')
+    records = [
+        {'title': 'The Meaning of Life', 'release_year': 1983},
+        {'title': 'Monty Python and the Holy Grail', 'release_year': 1975},
+        {'title': 'Life of Brian', 'release_year': 1979},
+        {
+            'title': 'And Now for Something Completely Different',
+            'release_year': 1971
+        },
+    ]
+    dataframe = pandas.DataFrame(records)
+
+    job = client.load_table_from_dataframe(dataframe, table_ref, location='US')
+
+    job.result()  # Waits for table load to complete.
+
+    assert job.state == 'DONE'
+    table = client.get_table(table_ref)
+    assert table.num_rows == 4
+    # [END bigquery_load_table_dataframe]
+    column_names = [field.name for field in table.schema]
+    assert 'title' in column_names
+    assert 'release_year' in column_names
```

Comment:
I think checking the columns is a good way to check.

I'm curious what the round-trip looks like if you do a full round trip with `client.list_rows(table).to_dataframe()`. Does Pandas treat the `__index_0__` column as a real index or does it add a new third column?

If it doesn't treat it as an index, I'd consider that a feature request for `to_dataframe()` more so than this method, though.